application.py

```diff
@@ -193,12 +194,13 @@ async def unknown_exception_handler(request, exc):
 @data_service_app.middleware("http")
 async def add_x_request_id_response_header(request: Request, call_next):
     response = await call_next(request)
-    response.headers["X-Request-ID"] = json_logging.get_correlation_id()
+    response.headers["X-Request-ID"] = json_logging.get_correlation_id(request=request)
     return response
 
 
 @data_service_app.on_event("startup")
 def startup_event():
+    json_logging.CREATE_CORRELATION_ID_IF_NOT_EXISTS = True
     json_logging.CORRELATION_ID_GENERATOR = lambda: "data-service-" + str(uuid.uuid1())
```

Comment:
Looks like a new correlation id is generated on startup. 
But we want new correlation id pr. session (if client sends no id), not pr. startup.
Did I miss something?

Reply:
Correlation ID will be created only if there is a request. If the request does not contain it, json_logging will create one.

Reply:
Ok!